pkg/utils/reader.go

```diff
@@ -72,6 +89,64 @@ func InitFileReader() *FileReader {
 	return &FileReader{}
 }
 
+type msg struct {
+	Msg         string
+	DomainName  string
+	Hostname    string
+	HTTPStatus  int
+	Batch       string
+	Region      string
+	Latency     int
+	Az          string
+	Url         string
+	UserAgent   string
+	IPv4Address string
+	Port        int
+}
+
+var logMessages = []string{
+	"%s for DRA plugin %q failed. Plugin returned an empty list for supported versions",
+	"%s for DRA plugin %q failed. None of the versions specified %q are supported. err='%v'",
+	"Unable to write event '%v' (retry limit exceeded!)",
+	"Unable to start event watcher: '%v' (will not retry!)",
+	"Could not construct reference to: '%v' due to: '%v'. Will not report event: '%v' '%v' '%v'",
+}
+
+func replacePlaceholders(template string) string {
```

Comment:
You should add a comment above `replacePlaceholders` that describes what it does and/or gives an example.